### ISSUE TITLE
[Feature Request] Editor Enhancement: More arrow shapes, bi-directional, dropdown menu for quick selecting pre-defined templates, ...

### ISSUE THREAD

**Issue by NNTin** [ShareX/ShareX#4227]
At work I'm using Screenpresso. ShareX is phenomenal. It's highly customizable and very powerful. It can do a lot of different things and it integrates very well.

However at work when I'm creating simple pictures and doing simple documentation the lack of pre-defined templates hinders workflow.  
It would be good if there are some pre-defined templates that can be quickly switched between.

I prefer using ShareX but during work Screenpresso is better optimized for the work flow.

What is Screenpresso doing well?

Screenpresso has pre-configured templates:  
While you can achieve almost achieve the same result by customizing the colors it takes a while to customize it. Screenpresso offers some finished templates making quick documentation easy and fast:
* [Arrows in different colors, shape, option for bidirectional](https://user-images.githubusercontent.com/18033169/61291880-06c32800-a7d0-11e9-9703-55159cc4a367.png)  
* [Some default rectangles (all configurable in ShareX as well but it doesn't exist as templates)](https://user-images.githubusercontent.com/18033169/61291963-3f630180-a7d0-11e9-9839-1c6dd88488ee.png)
* [Text templates also exist...](https://user-images.githubusercontent.com/18033169/61292004-63264780-a7d0-11e9-9149-d762e419dbc6.png)
* [Same goes for speech bubble...](https://user-images.githubusercontent.com/18033169/61292021-6d484600-a7d0-11e9-8d5d-ff9198e9ed03.png)
* [Different markers are pretty nice](https://user-images.githubusercontent.com/18033169/61292037-79340800-a7d0-11e9-97d8-51a8c209a813.png)

Extra options:
* [Magnifying glass](https://user-images.githubusercontent.com/18033169/61292174-d334cd80-a7d0-11e9-8c2a-8aff73b8d64a.png): I don't personally use this but I thought it was neat to point out Screenpresso has this.
* [Fancy border](https://user-images.githubusercontent.com/18033169/61292269-1131f180-a7d1-11e9-8dc0-d7ffa6e9465f.png). It looks incredibly good when you are taking pictures of code and pointing out some issues.

Proposal:
* Offer some templates for different arrow shapes (uni- and bi-directional, plain looking arrow and some "fancier" looking arrow)  
* Offer some templates for different text (font, color, border, ...)
* Offer a quick menu for selecting "default" colors instead of only pulling up the highly customizable color menu).
* Offer [fancy border](https://user-images.githubusercontent.com/18033169/61292269-1131f180-a7d1-11e9-8dc0-d7ffa6e9465f.png). I think a significant amount of people who are using ShareX's editor are programmers so this would be a nice addition.

ShareX is a phenomenal (and free!) product but it is lacking a bit of user experience. It is highly customizable but sometimes "less is more". Having a "dumbed down version" with pre-defined templates (while keeping the option to customize more) is often times better than only having the possibility to customize it. 

Take for instance the feature of highlighting a region in a certain color.   
In ShareX you currently select highlighting then you select color and a color menu pops up. There you can either select a color from the color wheel which will result in a color that is not consistent to my previous images or I can select one of the 32 default colors of which only 4 are "good looking". I can also pick from the recent colors. Then I have the explicitly close the window.  
In Screenpresso when you select highlighting a fancy dropdown menu pops up with 4 pre-defined colors. I can start immediately coloring without closing the dropdown menu (it automatically closes) or I can select a color from there. If I'm not satisfied with the color I can still change the color of the object by rightclicking. I can also set a new pre-defined template.  

TL;DR: The quick access of pre-defined templates of screenpresso means less time editing the image which increases productivity. It also has some neat looking arrows and other shapes ("code rectangle") which ShareX does not have which makes the end result look better. It would be great if these could be implemented into ShareX.

[This image was created in Screenpresso using the default templates.](https://user-images.githubusercontent.com/18033169/61295100-991afa00-a7d7-11e9-8498-ce0cbdc0416c.png)

**Comment by ShowGoShane** [comment]
Great suggestions!

**Comment by jacquen** [comment]
Good idea, Jaex, any plan for this?

**Comment by rainbowjose** [comment]
Also take a look at Monosnap editor. It much more friendly, but very good(in my opinion - simply best) example.

**Comment by 116581658** [comment]
I would like at least for the Speech balloon to be possible to round it corners.

**Comment by MoosMas** [comment]
Great suggestions, hopefully they will be implemented one day

**Comment by odyskat** [comment]
Hi I really like ShareX but have been using sniptool 2.1 just for the double-headed arrows (black hand mouse position on supplied screenshot)  (I mark dimensions in screenshots).
Also, there is a tool (red circle) that select a part of an image does a copy and allows me to magnify it (I use it as a "detail" view of an images' part) again quite useful. Both would be great additions. 

![image](https://github.com/ShareX/ShareX/assets/114591654/9c02a1a4-0e71-4658-bc2e-e1b6ecd8f3f8)

**Comment by szegi** [comment]
[Arrows in different colors, shape, option for bidirectional] would be great
